### PR TITLE
flif, new recipe

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

File: media-libs/flif/flif-20181126.recipe [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+lossy encoding of the entire image."
+HOMEPAGE="https://flif.info/"
+COPYRIGHT="2016 Jon Sneyers
+	Pieter Wuille"
+LICENSE="Apache v2
+	GNU GPL v3
+	GNU LGPL v3"
+REVISION="1"
+srcGitRev="74ea92bf1ab2db18ae6d2a521dff946fcc679618"
+SOURCE_URI="https://github.com/FLIF-hub/FLIF/archive/74ea92bf1ab2db18ae6d2a521dff946fcc679618.tar.gz"
```
Comment (alaviss) [inline media-libs/flif/flif-20181126.recipe]:
```suggestion
SOURCE_URI="https://github.com/FLIF-hub/FLIF/archive/$srcGitRev.tar.gz"
```

File: media-libs/flif/flif-20181126.recipe [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	# move headers
+	mkdir -p $includeDir
+	mv $prefix/include/* $includeDir
+	rm -rf $prefix/include
+
+	# move libs for secondaryArchSuffix
+	if [ "$targetArchitecture" = x86_gcc2 ]; then
+	mv $prefix/lib $prefix/lib2
+	mkdir $(dirname $libDir)
+	mv $prefix/lib2 $libDir
```
Comment (alaviss) [inline media-libs/flif/flif-20181126.recipe]: Please indent the interior of this `if` statement.

File: media-libs/flif/flif-20181126.recipe (hunk 1 above)
Comment (Begasus) [reply media-libs/flif/flif-20181126.recipe]: Jikes, forgot to adjust that one :)

**Comment by Begasus** [comment]
If no one objects this could be merged ...

File: media-libs/flif/flif-20181126.recipe [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+COPYRIGHT="2016 Jon Sneyers
+	Pieter Wuille"
+LICENSE="Apache v2
+	GNU GPL v3
+	GNU LGPL v3"
+REVISION="1"
+srcGitRev="74ea92bf1ab2db18ae6d2a521dff946fcc679618"
+SOURCE_URI="https://github.com/FLIF-hub/FLIF/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="aeeb4d24f19204f2b0cecc41ee1ebdda1c1ae730f0c6fa4969f67aab7e5eccac"
+SOURCE_DIR="FLIF-$srcGitRev"
```
[the remark this answers is not in the document]
Comment (Begasus) [reply media-libs/flif/flif-20181126.recipe]: Need to take a closer look into cmake in the src directory, will get back on that later :)

[the remark this answers is not in the document]
Comment (Begasus) [reply media-libs/flif/flif-20181126.recipe]: Switched to cmake, still need to move the headers and libs for secondary architecture (libs*)
Test fails 3 out of 5

File: media-libs/flif/flif-20181126.recipe [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	cd src
+	cmake . $cmakeDirArgs
+	sed -i 's/;-ltdl//g' CMakeCache.txt
+	sed -i 's/;ltdl//g' CMakeCache.txt
```
Comment (alaviss) [inline media-libs/flif/flif-20181126.recipe]: I would be more curious on what exactly generated this, but this should be solvable by adding `devel:libltdl` to `BUILD_REQUIRES`.

Comment (Begasus) [reply media-libs/flif/flif-20181126.recipe]: Seems I typed something wrong when searching for this :)

Comment (Begasus) [reply media-libs/flif/flif-20181126.recipe]: Fixed, seems it also needed the lib itself in the recipe :)

**Comment by Begasus** [comment]
@alaviss could you check?

**Comment by Begasus** [comment]
Created a descent patchset (thanks @alaviss ! ) :)
Also put the viewer (requires SDL2) in a seperate package tools
Test still fail 3 out of 5 (not sure if anyone is really looking for this) :)

**Pull request merged**